ably/types/mixins.py

```
@@ -46,10 +67,42 @@ def decode(data, encoding='', cipher=None):
                 data = json.loads(data)
-            elif encoding == 'base64' and isinstance(data, bytes):
-                data = bytearray(base64.b64decode(data))
             elif encoding == 'base64':
-                data = bytearray(base64.b64decode(data.encode('utf-8')))
+                data = bytearray(base64.b64decode(data)) if isinstance(data, bytes) \
+                    else bytearray(base64.b64decode(data.encode('utf-8')))
+                if not encoding_list: last_payload = data
+            elif encoding == ENC_VCDIFF:
+                if not context or not context.vcdiff_decoder:
+                    log.error('Message cannot be decoded as no VCDiff decoder available')
+                    raise AblyException('VCDiff decoder not available', 40019, 40019)
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)